src/app/system/manage-surveys/manage-surveys.component.ts

```diff
@@ -0,0 +1,68 @@
+/** Angular Imports */
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { MatTableDataSource, MatPaginator, MatSort } from '@angular/material';
+import { ActivatedRoute } from '@angular/router';
+
+@Component({
+  selector: 'mifosx-manage-surveys',
+  templateUrl: './manage-surveys.component.html',
+  styleUrls: ['./manage-surveys.component.scss'],
+})
+export class ManageSurveysComponent implements OnInit {
+  surveysData: any;
+  displayedColumns: string[] = ['key', 'name', 'description', 'countryCode', 'status', 'action'];
+  dataSource: MatTableDataSource<any>;
+
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+  @ViewChild(MatSort) sort: MatSort;
+
+  /**
+   * Retrieves the codes data from `resolve`.
+   * @param {ActivatedRoute} route Activated Route.
+   */
+  constructor(private route: ActivatedRoute) {
+    this.route.data.subscribe(( data: { surveys: any }) => {
+      this.surveysData = data.surveys;
+    });
+  }
+
+
+  /**
+   * Returns whether an element is active based on its active dates.
+   * @param {number} validFrom Date valid from
+   * @param {number} validTo Date valid to
+   */
+  isActive = function( validFrom: number, validTo: number) {
```

Comment:
```suggestion
  isActive(validFrom: number, validTo: number) {
```

Is function keyword necessary?